.github/actions/add-to-board-v1/src/run.ts

```diff
@@ -0,0 +1,81 @@
+import type { Core, Github } from './types'
+import getProjectBoardID from './getProjectBoardID'
+import addIssueToBoard from './addIssueToBoard'
+import getProjectBoardFieldList from './getProjectBoardFieldList'
+import moveIssueToColumn from './moveIssueToColumn'
+
+export default async function run(core: Core, github: Github): Promise<void> {
+  try {
+    const issueUrl = core.getInput('issue-url', { required: true })
+    const projectNumber = parseInt(core.getInput('project-number'))
+    const columnName = core.getInput('column-name')
+
+    if (isNaN(projectNumber)) {
+      core.setFailed('`project-number` must be a number')
+      return
+    }
+
+    // the issue card ID as a result of adding the issue to the project board
+    const issueCardIDs: string[] = []
+    const { owner } = github.context.repo
+    let issueUrls = issueUrl.includes(',') ? JSON.parse(issueUrl) : issueUrl
+
+    if (!Array.isArray(issueUrls)) {
+      issueUrls = [issueUrls]
+    }
+
+    const [{ id: projectBoardID }, { fields }] = await Promise.all([
+      getProjectBoardID({ projectNumber, owner }),
+      getProjectBoardFieldList({ projectNumber, owner })
+    ])
+
+    for (const url of issueUrls) {
+      core.info(`\nAdding issue ${url} to project board ${projectNumber}`)
+
+      const { id: issueCardID } = await addIssueToBoard({
+        projectNumber,
+        owner,
+        issueUrl: url
+      })
+
+      core.info(`\nReceived issue card ID ${issueCardID}`)
+      issueCardIDs.push(issueCardID)
+    }
+
+    // Status is the default field name for the project board columns e.g. Backlog, In progress, Done etc
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const statusField = fields.find(
+      field => field.name.toLowerCase() === 'status'
+    )!
+
+    const column = statusField.options.find(
+      option => option.name.toLowerCase() === columnName.toLowerCase()
+    )
+
+    if (!column) {
```

Comment:
We should do this check first before trying to add everything to the board. Otherwise you'll add a bunch of tickets but then they won't be in the correct column